cobbler/items/menu.py

```diff
@@ -107,24 +107,7 @@ def children(self, value: List[str]):
         :param value: The value to set the children to.
         :raises TypeError: Raised in case the children of menu have the wrong type.
         """
-        if not isinstance(value, list):
-            raise TypeError("Field children of object menu must be of type list.")
-        if isinstance(value, list):
-            if not all(isinstance(x, str) for x in value):
-                raise TypeError(
-                    "Field children of object menu must be of type list and all items need to be menu "
-                    "names (str)."
-                )
-            self._children = []
-            for name in value:
-                menu = self.api.find_menu(name=name)
-                if menu is not None:
-                    self._children.append(name)
-                else:
-                    self.logger.warning(
-                        'Menu with the name "%s" did not exist. Skipping setting as a child!',
-                        name,
-                    )
+        self._children = value
```

Comment:
Why did you remove this code? I added it in the past so you can pass a list of names or objects. If this is correct, I would love if we could put the reasoning in a commit message.

Reply:
The only reason I removed all the code except for one line in children.setter was that in profile.py this code also consists of exactly one line. Initially, I wrote the menu.py code and now I decided to check whether the implementation of properties in it corresponds to the same properties in other classes. I thought that the code in profile.py was more correct, but in menu.py it was simply forgotten to fix it. The children property is used in tftpgen.py when forming submenus and it is important that it is implemented correctly. I haven't examined the item.get_children code, so I can't say which code is more correct. Revert the old code back?

Reply:
Since we have not type guarantees (after all we only add type hints to the code) and the old code contains the functionality the new code has, I would indeed vote for reverting it to the previous state. If you think you are absolutely certain that this code is not required then I gladly accept the changes your have proposed.

Reply:
I agree, I returned this code back.